bug report

`block_test.go`
```
func TestBug(t *testing.T) {
    // Reproduce the bug
    b := Block{}
    rlp.DecodeBytes([]byte("\xf8\xf9\xf8Ġ00000000000000000000000000000000\x8800000000\x8200\x9400000000000000000000\x82000\xa000000000000000000000000000000000\xa000000000000000000000000000000000\xa000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000"), &b)
}
```

#### Panic Backtrace
```
--- FAIL: TestBug (0.00s)
panic: runtime error: index out of range [0] with length 0 [recovered]
    panic: runtime error: index out of range [0] with length 0

goroutine 6 [running]:
testing.tRunner.func1.2({0x6cf5a0, 0xc00001e3f0})
    /usr/lib/golang/src/testing/testing.go:1396 +0x24e
testing.tRunner.func1()
    /usr/lib/golang/src/testing/testing.go:1399 +0x39f
panic({0x6cf5a0, 0xc00001e3f0})
    /usr/lib/golang/src/runtime/panic.go:884 +0x212
github.com/vechain/thor/block.(*extension).DecodeRLP(0xc0001c60d8, 0xc0001c60d8?)
    /home/oblivion/go/src/github.com/vechain/thor/block/extension.go:56 +0x2b6
github.com/ethereum/go-ethereum/rlp.decodeDecoderNoPtr(0x6d98a0?, {0x6b3d20?, 0xc0001c60d8?, 0x830080?})
    /home/oblivion/go/pkg/mod/github.com/vechain/go-ethereum@v1.8.15-0.20220606031836-4784dac628d7/rlp/decode.go:538 +0x62
github.com/ethereum/go-ethereum/rlp.makeStructDecoder.func1(0xc000030500, {0x6d98a0?, 0xc0001c6000?, 0x0?})
    /home/oblivion/go/pkg/mod/github.com/vechain/go-ethereum@v1.8.15-0.20220606031836-4784dac628d7/rlp/decode.go:442 +0x8ec
github.com/ethereum/go-ethereum/rlp.(*Stream).Decode(0xc000073990?, {0x68c3a0?, 0xc0001c6000?})
    /home/oblivion/go/pkg/mod/github.com/vechain/go-ethereum@v1.8.15-0.20220606031836-4784dac628d7/rlp/decode.go:810 +0x205
github.com/vechain/thor/block.(*Header).DecodeRLP(0xc0001a46e0, 0xc0001a46e0?)
    /home/oblivion/go/src/github.com/vechain/thor/block/header.go:247 +0x5e
github.com/ethereum/go-ethereum/rlp.decodeDecoderNoPtr(0x6b0400?, {0x6b3b40?, 0xc0001a46e0?, 0x830080?})
    /home/oblivion/go/pkg/mod/github.com/vechain/go-ethereum@v1.8.15-0.20220606031836-4784dac628d7/rlp/decode.go:538 +0x62
github.com/ethereum/go-ethereum/rlp.makeStructDecoder.func1(0xc000030500, {0x6b0400?, 0xc0001a46e0?, 0x595e5a?})
    /home/oblivion/go/pkg/mod/github.com/vechain/go-ethereum@v1.8.15-0.20220606031836-4784dac628d7/rlp/decode.go:442 +0x8ec
github.com/ethereum/go-ethereum/rlp.(*Stream).Decode(0xc000030500?, {0x68f420?, 0xc0001a46e0?})
    /home/oblivion/go/pkg/mod/github.com/vechain/go-ethereum@v1.8.15-0.20220606031836-4784dac628d7/rlp/decode.go:810 +0x205
github.com/vechain/thor/block.(*Block).DecodeRLP(0xc0000a15f0, 0xc0000a15f0?)
    /home/oblivion/go/src/github.com/vechain/thor/block/block.go:85 +0x54
github.com/ethereum/go-ethereum/rlp.decodeDecoderNoPtr(0x6c9f20?, {0x6bd720?, 0xc0000a15f0?, 0xc000062638?})
    /home/oblivion/go/pkg/mod/github.com/vechain/go-ethereum@v1.8.15-0.20220606031836-4784dac628d7/rlp/decode.go:538 +0x62
github.com/ethereum/go-ethereum/rlp.(*Stream).Decode(0xc000030500?, {0x6c9f20?, 0xc0000a15f0?})
    /home/oblivion/go/pkg/mod/github.com/vechain/go-ethereum@v1.8.15-0.20220606031836-4784dac628d7/rlp/decode.go:810 +0x205
github.com/ethereum/go-ethereum/rlp.DecodeBytes({0xc000010800, 0xfb, 0xfb}, {0x6c9f20, 0xc0000a15f0})
    /home/oblivion/go/pkg/mod/github.com/vechain/go-ethereum@v1.8.15-0.20220606031836-4784dac628d7/rlp/decode.go:139 +0xbf
github.com/vechain/thor/block_test.TestBug(0x0?)
    /home/oblivion/go/src/github.com/vechain/thor/block/block_test.go:153 +0x8c
testing.tRunner(0xc000196820, 0x7dcc40)
    /usr/lib/golang/src/testing/testing.go:1446 +0x10b
created by testing.(*T).Run
    /usr/lib/golang/src/testing/testing.go:1493 +0x35f
FAIL    github.com/vechain/thor/block   0.011s
FAIL
```

Found by https://github.com/vechain/thor/pull/567

--- a/block/extension.go
+++ b/block/extension.go
@@ -49,8 +49,8 @@ func (ex *extension) DecodeRLP(s *rlp.Stream) error {
 		}
 	}
 
-	if len(raws) > 2 {
-		return errors.New("rlp: too much elements in extension")
+	if len(raws) == 0 || len(raws) > 2 {
+		return errors.New("rlp: unexpected extension")
 	} else {
 		var alpha []byte
 		if err := rlp.DecodeBytes(raws[0], &alpha); err != nil {

